app/src/pages/Lobby/index.tsx

```diff
@@ -94,16 +94,22 @@ function Lobby() {
         <>
           <div className={classes.section}>
             <SettingsSection
-              cardPlayStyle={cardPlayStyle}
+              cardPlayStyle={currentGame.card_play_style}
               setCardPlayStyle={(value) => {
-                setCardPlayStyle(value as GameCardPlayStyleEnum)
+                // setCardPlayStyle(value as GameCardPlayStyleEnum)
                 updateGameSettings({
                   variables: {
                     id: currentGame.id,
                     input: {
                       card_play_style: value as GameCardPlayStyleEnum,
                     },
                   },
+                  optimisticResponse: {
+                    update_games_by_pk: {
+                      ...currentGame,
```

Comment:
I wonder if two fragments would help / minimize the scope of these setting cache updates, i.e.

1. `fragment GameSettings on games` with the (base) [`UpdateGameSettings` properties](https://github.com/avimoondra/fishbowl/blob/a9675dba5e8780fa535fc8ff54ca78c173136871/app/src/pages/Lobby/operations.graphql#L3-L10)

1. a full fragment that "extends" from that one:

    ```graphql
    fragment Game on games {
        ...GameSettings
        state
        (etc)
    ```

    with the rest of the [`CurrentGame` properties](https://github.com/avimoondra/fishbowl/blob/a9675dba5e8780fa535fc8ff54ca78c173136871/app/src/graphql/operations.graphql#L31-L64)

Reply:
Most definitely 👍 - there's no use of fragments across this whole codebase yet! Bout damn time.